Error compiling connection-0.3.0

## Issue

When I try to compile connection (a dependency of xmobar, which is what I am trying to install), I get a compilation error.

## Machine info

```
$ uname -a
Linux kuro 4.14.83-gentoo #6 SMP Fri Feb 15 20:45:56 -00 2019 x86_64 Intel(R) Core(TM) i7-4600U CPU @ 2.10GHz GenuineIntel GNU/Linux
$ ghc --version
The Glorious Glasgow Haskell Compilation System, version 8.0.2
$ cabal --version
cabal-install version 1.24.0.2
compiled using version 1.24.2.0 of the Cabal library
```

## Build log

```
Build log ( /home/lattis/.cabal/logs/connection-0.3.0.log ):
cabal: Entering directory '/tmp/cabal-tmp-18567/connection-0.3.0'
Configuring connection-0.3.0...
Building connection-0.3.0...
Preprocessing library connection-0.3.0...
[1 of 2] Compiling Network.Connection.Types ( Network/Connection/Types.hs, dist/build/Network/Connection/Types.o )
[2 of 2] Compiling Network.Connection ( Network/Connection.hs, dist/build/Network/Connection.o )

Network/Connection.hs:213:22: error:
    • No instance for (Read PortNumber) arising from a use of ‘reads’
    • In the expression: reads portS
      In the expression:
        case reads portS of {
          [(sPort, "")] -> Just (sHost, sPort)
          _ -> Nothing }
      In a case alternative:
          (sHost, ':' : portS)
            -> case reads portS of {
                 [(sPort, "")] -> Just (sHost, sPort)
                 _ -> Nothing }
cabal: Leaving directory '/tmp/cabal-tmp-18567/connection-0.3.0'
```

## Steps to reproduce

```
$ cabal install xmobar
```

--- a/connection.cabal
+++ b/connection.cabal
@@ -27,7 +27,7 @@ Library
                    , bytestring
                    , containers
                    , data-default-class
-                   , network >= 2.6
+                   , network >= 2.6.3
                    , tls >= 1.4
                    , socks >= 0.6
                    , x509 >= 1.5

